tmk_core/common/progmem.h

```diff
@@ -5,9 +5,9 @@
 #   include <avr/pgmspace.h>
 #else
 #   define PROGMEM
-#   define pgm_read_byte(p)     *((unsigned char*)p)
-#   define pgm_read_word(p)     *((uint16_t*)p)
-#   define pgm_read_dword(p)    *((uint32_t*)p)
+#   define pgm_read_byte(p)     *((unsigned char*)(p))
+#   define pgm_read_word(p)     *((uint16_t*)(p))
+#   define pgm_read_dword(p)    *((uint32_t*)(p))
```

Comment:
The layer macro changes are fine, but this is potentially breaking for non-AVR boards. The thing is that this macro has been without parentheses, and thus with different behavior from the AVR version, for a while. It's possible that some ARM boards were made to work with it in its “broken” state, and changing this might break them. We need to look into this further.

Reply:
Looking through the code, there is only a couple of instances of where this may be misinterpreted.  

So, it looks like it could be safely merged.

Though, I'd like to see a ATSAM, ChibiOS and AVR device tested, first. 

I can do 2 out of 3 there.